packages/component/src/Utils/TypeFocusSink/Box.js

```diff
@@ -0,0 +1,93 @@
+import memoize from 'memoize-one';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import Context from './Context';
+import getTabIndex from './getTabIndex';
+import inputtableKey from './inputtableKey';
+
+const DEFAULT_STYLE = { outline: 0 };
+
+export default class Box extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.focus = this.focus.bind(this);
+    this.handleKeyDownCapture = this.handleKeyDownCapture.bind(this);
+
+    this.createContext = memoize((staticContext, sendFocusRef = React.createRef()) => ({
+      ...staticContext,
+      sendFocusRef
+    }));
+
+    this.state = {
+      context: {}
+    };
+  }
+
+  focus() {
+    const { current } = this.props.sendFocusRef;
+
+    current && current.focus();
+  }
+
+  handleKeyDownCapture(event) {
+    const target = event.target;
+    const tabIndex = getTabIndex(target);
+
+    if (
+      event.altKey
+      || (event.ctrlKey && event.key !== 'v')
+      || event.metaKey
+      || (!inputtableKey(event.key) && event.key !== 'Backspace')
+    ) {
+      // Ignore if one of the utility key (except SHIFT) is pressed
+      // E.g. CTRL-C on a link in one of the message should not jump to chat box
+      // E.g. "A" or "Backspace" should jump to chat box
+      return;
+    }
+
+    if (
+      typeof tabIndex !== 'number'
+      || tabIndex < 0
+    ) {
+      event.stopPropagation();
+
+      // let key;
+
+      // Quirks: onKeyDown we re-focus, but the newly focused element does not receive the subsequent onKeyPress event
+      //         It is working in Chrome/Firefox/IE, confirmed not working in Edge/16
+      //         So we are manually appending the key if they can be inputted in the box
+      // if (/(^|\s)Edge\/16\./.test(window.navigator.userAgent)) {
+      //   key = inputtableKey(event.key);
+      // }
```

Comment:
delete? is it working now?

Reply:
No, it doesn't work. It's a gray area.

Removed the code as this will be hard to maintain later and not much UX value.